crates/net/net.rs

```diff
@@ -231,18 +233,38 @@ async fn serve_requests(tcp_addr: SocketAddr, signer: SigningKey) {
     let mut pending_conn = client.decode_ack_message(&secret_key, msg, auth_data);
     info!("Completed handshake!");
 
-    let hello_msg = RLPxMessage::Hello(
-        SUPPORTED_CAPABILITIES.to_vec(),
+    let hello_msg = RLPxMessage::Hello(p2p::HelloMessage::new(
+        SUPPORTED_CAPABILITIES
+            .into_iter()
+            .map(|(name, version)| (name.to_string(), version))
+            .collect(),
         PublicKey::from(signer.verifying_key()),
-    );
+    ));
 
     pending_conn.send(hello_msg, &mut stream).await;
 
-    let mut conn = pending_conn.receive_hello(&mut stream).await;
+    // Receive Hello message
+    let mut conn = pending_conn.receive(&mut stream).await;
 
     info!("Completed Hello roundtrip!");
 
-    conn.send(RLPxMessage::Ping(), &mut stream).await;
+    // Send Ping
+    conn.send(RLPxMessage::Ping(p2p::PingMessage::new()), &mut stream)
+        .await;
+
+    // Receive three messages
+    // TODO implement listen loop instead
+    conn.receive(&mut stream).await;
```

Comment:
This was just for testing purposes... Normally those are the Ping response (Pong), some eth message (when the capability is enabled and shared) and I left some other receive just in case for debug reasons.

As the comment says, this is just temporary and we need to implement the proper listen loop